### PR TITLE
Eiffel curl updated:

### PR DESCRIPTION
**Pull request by jvelilla**
Added support to send mime emails.
Added example to show how to use it.

### CODE REVIEW

File: Src/examples/cURL/smtp_mime/smtp_mime.ecf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+<system xmlns="http://www.eiffel.com/developers/xml/configuration-1-16-0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:schemaLocation="http://www.eiffel.com/developers/xml/configuration-1-16-0 http://www.eiffel.com/developers/xml/configuration-1-16-0.xsd" name="smtp_mime">
+	<target name="smtp_mime">
+		<root cluster="smtp_mime" class="APPLICATION" feature="make"/>
+		<option warning="true" is_obsolete_routine_type="true" syntax="transitional">
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/smtp_mime.ecf]: I guess you can remove the part ` is_obsolete_routine_type="true" syntax="transitional" `

File: Src/library/cURL/curl_opt_constants.e [hunk 2]
```diff
@@ -801,6 +801,16 @@ feature -- Protocol
 			is_class: class
 		end
 
+	curlopt_mimepost: INTEGER
+			-- Declared as CURLOPT_TIMEVALUE
```
Comment (jocelyn) [inline Src/library/cURL/curl_opt_constants.e]: the comment mentions CURLOPT_TIMEVALUE ... but I guess this is a copy/paste error as it should mention CURLOPT_MIMEPOST

File: Src/library/cURL/interface/curl_easy_externals_i.e [hunk 3]
```diff
@@ -115,6 +115,16 @@ feature -- Command
 			setopt_void_star (a_curl_handle, a_opt, a_file.file_pointer)
 		end
 
+	setopt_mime (a_curl_handle: POINTER; a_opt: INTEGER; a_mime: POINTER)
+			-- Declared as curl_easy_setopt().
```
Comment (jocelyn) [inline Src/library/cURL/interface/curl_easy_externals_i.e]: Not sure if this is a typo or not about "curl_easy_setopt()"
but it is, then the previous features have the same typo.

File: Src/examples/cURL/smtp_mime/application.e [hunk 4]
```diff
@@ -0,0 +1,528 @@
...
+				-- add the from email.
+			l_from := "sender@example.org"
+			l_to := "addressee@example.net"
+			l_cc := "info@example.org"
+
+			create l_message.make_from_string (message)
+			l_message.replace_substring_all ("$TO", l_to)
+			l_message.replace_substring_all ("$CC", l_cc)
+			l_message.replace_substring_all ("$FROM", l_from)
+			if curl_easy.is_dynamic_library_exists then
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/application.e:49]: each time I see this `is_dynamic_library_exists`  I find the name bad .. should be `dynamic_library_exists` or even `dynamic_library_available` 
but I guess too late to change.

File: Src/examples/cURL/smtp_mime/application.e [hunk 5]
```diff
@@ -0,0 +1,528 @@
...
+				headers := curl_externals.slist_append (headers, "Message-ID: <dcd7cb36-11db-487a-9f3a-e652a9458efd@rfcpedant.example.org>" )
+				headers := curl_externals.slist_append (headers, "Subject: example sending a MIME-formatted message" )
+				curl_easy.setopt_slist (curl_handle, {CURL_OPT_CONSTANTS}.curlopt_httpheader, headers)
+
+
+					 -- Build the mime message.
+				mime := curl_externals.curl_mime_init (curl_handle)
+
+				    -- The inline part is an alterative proposing the html and the text versions of the e-mail.
+    			alt	:= curl_externals.curl_mime_init (curl_handle)
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/application.e]: bad indentation

File: Src/examples/cURL/smtp_mime/application.e [hunk 6]
```diff
@@ -0,0 +1,528 @@
...
+class
+	APPLICATION
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make
+			-- Run application.
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/application.e]: maybe you can find a better comment.

**Review by jocelyn — CHANGES_REQUESTED** [review]

File: Src/examples/cURL/smtp_mime/smtp_mime.ecf (hunk 1 above)
Comment (jvelilla) [reply Src/examples/cURL/smtp_mime/smtp_mime.ecf]: done

File: Src/library/cURL/curl_opt_constants.e (hunk 2 above)
Comment (jvelilla) [reply Src/library/cURL/curl_opt_constants.e]: yes, fixed.

File: Src/library/cURL/interface/curl_easy_externals_i.e (hunk 3 above)
Comment (jvelilla) [reply Src/library/cURL/interface/curl_easy_externals_i.e]: Not sure, all the features calling the external C function used the same convention, maybe to indicate that was C function, I've updated all the comments.

File: Src/examples/cURL/smtp_mime/application.e (hunk 5 above)
Comment (jvelilla) [reply Src/examples/cURL/smtp_mime/application.e]: fixed

File: Src/examples/cURL/smtp_mime/application.e (hunk 6 above)
Comment (jvelilla) [reply Src/examples/cURL/smtp_mime/application.e]: done

File: Src/library/cURL/interface/curl_externals_i.e [hunk 7]
```diff
@@ -253,9 +354,158 @@ feature {NONE} -- C externals
...
+		require
+			a_api_exists: not a_api.is_default_pointer
+			a_curl_handle_exists: not a_curl_handle.is_default_pointer
+		external
+			"C inline use <curl/curl.h>"
+		alias
+			"[
+				#ifdef CURL_STATICLIB
+				(FUNCTION_CAST(void , (curl_mime *)) $a_api)
+										((curl_mime *)$a_curl_handle);
```
Comment (jocelyn) [inline Src/library/cURL/interface/curl_externals_i.e]: bad indentation

File: Src/library/cURL/interface/curl_externals_i.e [hunk 8]
```diff
@@ -253,9 +354,158 @@ feature {NONE} -- C externals
...
+		end
+
+	c_mime_filedata (a_api: POINTER; a_curl_handle: POINTER; a_filename: POINTER): INTEGER
+			-- CURLcode curl_mime_filedata(curl_mimepart * part , const char * filename );
+		require
+			a_api_exists: not a_api.is_default_pointer
+			exists_mimepart: not a_curl_handle.is_default_pointer
+		external
+				"C inline use <curl/curl.h>"
+			alias
```
Comment (jocelyn) [inline Src/library/cURL/interface/curl_externals_i.e]: bad indentation for the whole alias block

File: Src/library/cURL/interface/curl_easy_externals_i.e [hunk 9]
```diff
@@ -355,7 +365,7 @@ feature {NONE} -- C externals
 
 	c_getinfo (a_api: POINTER; a_curl_handle: POINTER; a_opt: INTEGER; a_data: POINTER): INTEGER
 			-- C implementation of `curl_easy_getinfo'.
-			-- Declared as curl_easy_setopt ().
+			-- Declared as curl_easy_setopt .
```
Comment (jocelyn) [inline Src/library/cURL/interface/curl_easy_externals_i.e:368]: is it really "Declared as curl_easy_setopt"?? setopt ?
in fact I don't understand this comment here, and elsewhere.

File: Src/examples/cURL/smtp_mime/application.e [hunk 10]
```diff
@@ -0,0 +1,524 @@
...
+			slists: POINTER
+			curl_externals: CURL_EXTERNALS
+			l_message: STRING
+			mime: POINTER
+			alt: POINTER
+			part: POINTER
+		do
+			io.put_string ("Eiffel cURL SMTP example showing how to send e-mails with mime.")
+			io.put_new_line
+			print ("%NLibCurl Veriosn:" + (create {CURL_OPT_CONSTANTS}).libcurl_version.out)
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/application.e]: typo "Veriosn"...

File: Src/examples/cURL/smtp_mime/application.e [hunk 11]
```diff
@@ -0,0 +1,524 @@
+﻿note
+	description: "[
+					cURL https://curl.haxx.se/libcurl/c/smtp-mail.html
```
Comment (jocelyn) [inline Src/examples/cURL/smtp_mime/application.e]: remove a few left indentations

**Review by jocelyn — APPROVED** [review]
a few things to fix, but LGTM

File: Src/library/cURL/interface/curl_externals_i.e (hunk 8 above)
Comment (jvelilla) [reply Src/library/cURL/interface/curl_externals_i.e]: Fixed

**Pull request closed without merging**